pystiche_papers/sanakoyeu_et_al_2018/_modules.py

```diff
@@ -273,3 +276,45 @@ def transformer(style: Optional[str] = None,) -> Transformer:
 
     """
     return Transformer()
+
+
+class Discriminator(pystiche.Module):
+    r"""Discriminator from :cite:`SKL+2018`.
+
+    Args:
+        in_channels: Number of channels in the input. Defaults to ``3``.
+    """
+
+    def __init__(self, in_channels: int = 3) -> None:
+        def discriminator_encoder_modules(
+            in_channels: int = 3, inplace: bool = True
+        ) -> List[nn.Sequential]:
+            conv_block = partial(
+                ConvBlock,
+                kernel_size=5,
+                stride=2,
+                padding="same",
+                act="lrelu",
+                inplace=inplace,
+            )
+
+            return channel_progression(
+                lambda in_channels, out_channels: conv_block(in_channels, out_channels),
```

Comment:
1. `channel_progression` expects a callable with the arguments `in_channels` and `out_channels`. Thus, simply passing `conv_block` would be sufficient.
2. Since we have `channel_progression` now, `partial` is no longer needed. Simply pass a `lambda` callable of `ConvBlock`.